### DISCUSSION TITLE
Webhook Error When Trying to Update Embed

### DISCUSSION THREAD

**Discussion by tropicthedev** [discord-jda/JDA#2709] [Questions and Help] [answered]
Hi, I am trying to update an embed after certain criteria have been met but I keep running into an error that I am unsure of how to fix. 

[code-extract.txt](https://github.com/user-attachments/files/16398269/code-extract.txt)

 It errors out on the last few lines where I actually edit the message with the error being `java.lang.IllegalStateException: Sending a webhook request requires the interaction to be acknowledged before expiration` while referencing the 

```java
event.getMessage().editMessageEmbeds(
                   embedBuilder.build()
           ).setActionRow(
                   Button.primary(ButtonIds.JOIN_THREAD, "Join Tread")
                           .withEmoji(Emoji.fromFormatted("🚀")),
                   Button.danger(ButtonIds.BAN, "Ban Applicant (Coming Soon)")
                           .withEmoji(Emoji.fromFormatted("🦵"))
                           .asDisabled()
           ).queue();
```

block

**Comment by tropicthedev** [comment]
I added the code extract in a txt file because for some reason GitHub could not format the code block properly.

**Comment by MinnDevelopment** [comment, marked as answer]
All interactions must use `event.reply()` or `event.deferReply()` to send messages. In the future, include the full context, including what event you're using when posting a question.

**Reply by tropicthedev** [reply]
Thank you. In the file I had the event in the method parameters, but I will be sure to make it clear the next time.

I used these methods to defer the message edit and then edit the embeds later in the file.

`event.deferEdit().queue();`
`event.getMessage().editMessageEmbeds()`
